### PR TITLE
remove path a file

### PR DESCRIPTION
**Pull request by md-miran**
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/blobber/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master BEFORE merging this PR

Associated PRs (Link as appropriate):
- zbox:
- zwalelet:
- gosdk:
- 0chain:
- system_test: 

Issue link: https://github.com/0chain/blobber/issues/454
we will throw error when upload remote path is a file.
Remote filePath = `/a/b.txt` 
Check if `/a` is a file or dir. If it is a file, throw error.

### CODE REVIEW

File: code/go/0chain.net/blobbercore/handler/object_operation_handler.go [hunk 1]
```diff
@@ -1169,3 +1175,24 @@ func getExistingFileRef(fsh *StorageHandler, ctx context.Context, r *http.Reques
...
+
+// check upload path is a file
+// ex := /path1/path2/path3/name
+// checking if /path1/path2/path3 is a file or dir
+func isUploadPathAFile(ctx context.Context, r *http.Request, allocationObj *allocation.Allocation, fileOperation string) bool {
+	if fileOperation == constants.FileOperationInsert {
+		var formData allocation.UpdateFileChanger
+		uploadMetaString := r.FormValue(getFormFieldName(fileOperation))
+		err := json.Unmarshal([]byte(uploadMetaString), &formData)
+		if err == nil {
```
Comment (lpoli) [inline code/go/0chain.net/blobbercore/handler/object_operation_handler.go:1099]: This return of `false` when err != nil is erroneous logically

**Review by lpoli — CHANGES_REQUESTED** [review]
This checking need no extra function.
Simply get a reference and check its `refType`

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | f_miran_blobber_file_path  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

File: code/go/0chain.net/blobbercore/handler/object_operation_handler.go [hunk 2]
```diff
@@ -1081,3 +1087,24 @@ func getExistingFileRef(fsh *StorageHandler, ctx context.Context, r *http.Reques
...
+			fileDir := filepath.Dir(formData.Path)
+			if fileDir != "/" && fileDir != "" {
+				dirReference, _ := reference.GetReference(ctx, allocationObj.ID, fileDir)
```
Comment (cnlangzi) [inline code/go/0chain.net/blobbercore/handler/object_operation_handler.go:1102]: it is not worth to get whole reference record. it is enough to  read  `type` only.

**Review by cnlangzi — CHANGES_REQUESTED** [review]

**Pull request closed without merging**